src/Disks/ObjectStorages/Web/WebObjectStorage.cpp

```diff
@@ -124,7 +124,19 @@ bool WebObjectStorage::exists(const StoredObject & object) const
 bool WebObjectStorage::exists(const std::string & path) const
 {
     LOG_TRACE(&Poco::Logger::get("DiskWeb"), "Checking existence of path: {}", path);
+    return tryGetFileInfo(path) != std::nullopt;
+}
+
+WebObjectStorage::FileData WebObjectStorage::getFileInfo(const String & path) const
+{
+    auto file_info = tryGetFileInfo(path);
+    if (!file_info)
+        throw Exception(ErrorCodes::LOGICAL_ERROR, "No such file: {}", path);
```

Comment:
Why is it a logical error? Do we have any guarantees that the file should be there, or what are our assumptions?

Reply:
yes, better to use `FILE_DOESNT_EXIST`.
In general it should be checked for `exists` method or be sure that file exists as it was a result of `list` or `iterateDirectory` methods.